src/Wangkanai.Detection/Version.cs

```diff
@@ -1,18 +1,26 @@
 // Copyright (c) 2016 Sarin Na Wangkanai, All Rights Reserved.
 // The GNU GPLv3. See License.txt in the project root for license information.
 
+using System;
+
 namespace Wangkanai.Detection
 {
     public class Version
     {
         public Version(string major, string minor)
         {
+            if (major == null) throw new ArgumentNullException(nameof(major));
```

Comment:
I think we can use here and below `string.IsNullOrEmpry()` and check for empty string too.